src/coffea/jetmet_tools/CorrectedMETFactory.py

```diff
@@ -4,11 +4,11 @@
 
 
 def corrected_polar_met(
-    met_pt, met_phi, jet_pt, jet_phi, jet_pt_orig, positive=None, dx=None, dy=None
+    met_pt, met_phi, jet_pt, jet_phi, positive=None, dx=None, dy=None
 ):
     sj, cj = numpy.sin(jet_phi), numpy.cos(jet_phi)
-    x = met_pt * numpy.cos(met_phi) + awkward.sum((jet_pt - jet_pt_orig) * cj, axis=1)
-    y = met_pt * numpy.sin(met_phi) + awkward.sum((jet_pt - jet_pt_orig) * sj, axis=1)
+    x = met_pt * numpy.cos(met_phi) + awkward.sum(jet_pt * cj, axis=1)
```

Comment:
Shouldn't this be calculating a difference in the corrected momentum with respect to each jet? That's what it says in the equation you quote, that's what the code was doing before. Now it does not. Please justify this change.

That or create a test that compares to outputs from some other tool.

Reply:
I have attached the example code in the [issue](https://github.com/CoffeaTeam/coffea/issues/1064).
The Type-I correction calculation formula has been implemented in `corrected_jets`：
```
def corrected_jets(self, jets, event_rho, lazy_cache):
        jets_L123 = self.jec_factory.build(
            add_jme_variables(jets, event_rho),
            lazy_cache #events.caches[0]
        )
        jets_L1 = self.jec_factory_L1.build(
            add_jme_variables(jets, event_rho),
            lazy_cache #events.caches[0]
        )
        emFraction = jets_L123.chEmEF + jets_L123.neEmEF
        mask_jec = (jets_L123.pt > 15) & (emFraction <= 0.9)
        selected_jets_L123 = ak.mask(jets_L123, mask_jec)
        jets_jec = selected_jets_L123
        jets_jec['pt'] = selected_jets_L123.pt - jets_L1.pt
        return jets_jec
```
Then propagate the difference value calculation of jet_pt corrected by L123 and L1 to MET through the following code:
```
jets_col = self._jmeu.corrected_jets(jets_col, rho, cache)
met  = self._jmeu.corrected_met(event.RawMET,event.MET, jets_col, event.fixedGridRhoFastjetAll, event.caches[0])
```
Does this explain that my updated code is correct?

Reply:
Ah this is a very clear to implement this since it now becomes unclear what the user is getting as a physics object. They expect to get the jets corrected by the stack that they put in, not something else that is suitable for MET correction.

Instead, it would be best to create two CorrectedJetsFactories:
```python3
jets_l123 = CorrectedJetFactory(config_L123).build(events.Jet) #add rho and whatnot to events.Jet
jets_l1 = CorrectedJetFactory(config_L1).build(events.Jet)

#then set ptRaw in the JECStack to jets_l1.pt
corr_met = CorrectedMetFactory(config_l123_l1).build(L1_met, jets_l123)
```

This would require no code changes to my understanding.

Reply:
I agree that set `ptRaw` to `jets_L1.pt`,it's a good alternative.
But the plus signs `+` here should be changed to `-`. Just like:
`x = met_pt * numpy.cos(met_phi) - awkward.sum((jet_pt - jet_pt_orig) * cj, axis=1)`